waiter/src/waiter/scheduler.clj

```diff
@@ -451,7 +461,10 @@
                                                  :failed-instances all-failed-instances
                                                  :scheduler-sync-time request-instances-time)])
                                         scheduler-messages)
-                  instance-id->unhealthy-instance' (pc/map-from-vals :id unhealthy-instances)
+                  instance-id->unhealthy-instance' (into {} (map (fn [{:keys [id] :as instance}]
+                                                                   (let [{:keys [flags]} (get instance-id->unhealthy-instance id)]
```

Comment:
super-nitpick: we can use `get-in`: `(let [flags (get-in instance-id->unhealthy-instance [id :flags])] ...)`